ddtrace/tracer/option.go

```diff
@@ -215,6 +218,14 @@ func WithDogstatsdAddress(addr string) StartOption {
 	}
 }
 
+// WithSamplingRules specifies the sampling rates to apply to spans based on the
+// provided rules.
+func WithSamplingRules(rules ...SamplingRule) StartOption {
+	return func(cfg *config) {
+		cfg.samplingRules = rules
```

Comment:
If we use the same struct (but also if not) we could compile the regexp here straight away